source/rfc/zuul-development-guidelines.rst

```diff
@@ -0,0 +1,24 @@
+Zuul job configuration development guidelines
+=============================================
+
+Zuul YAMLs - general
+
+#. Playbook naming: `playbooks/<job_name>/{run,pre,post}.yaml`
+#. Zuul-jobs vs project-config: which jobs/roles where
+#. Site-variables
+#. Nodes(ets) naming
+#. job variants - where to override settings
+
+Variables
+
+#. common libs usage
+#. Zuul return
+#. Docker tags
+#. Docker registries
+#. defaults and vars
+#. looking up project source paths
+
+Ansible code
```

Comment:
In my opinion, roles should be created as a part of pipeline - either as utilities (e.g. `install-something`, `configure-that`) or as parts of pipeline (`build-container`, `publish-container`) with a clear definition if role's inputs and outputs. Readme maybe? 
Inputs should be defined as defaults.